ogr/ogrsf_frmts/sosi/ogrsosilayer.cpp

```diff
@@ -315,6 +315,12 @@ OGRFeature *OGRSOSILayer::GetNextFeature() {
         case L_PUNKT: {  /* point */
             oGType = wkbPoint;
             OGRPoint *poPoint = poParent->papoBuiltGeometries[oNextSerial.lNr]->toPoint();
+            if (poPoint == nullptr) {
```

Comment:
what's the issue exactly ? That poParent->papoBuiltGeometries[oNextSerial.lNr] is nullptr, or toPoint() return nullptr ?  If the geometry provided to toPoint() is not null, then toPoint() shouldn't normally return nullptr (it will only do that in debug builds, since a dynamic_cast<> will be used, but in release builds as static_cast<> will be used), unless the geometry is not of type point. In which case this should be handled before calling toPoint() by testing the geometry type.

Reply:
Sure. Here is how to reproduce (on Linux/Unix):

```
wget https://transfer.sh/yJ1MKM/Plan-Kp3VnALLplanbase-T20220101_4205_Euref89_iso.sos
ogrinfo Plan-Kp3VnALLplanbase-T20220101_4205_Euref89_iso.sos polygons
```
The file should be available for two weeks from now...

Reply:
ok, your patch looks reasonable.
If might also be prudent to enhance the check to
if (poParent->papoBuiltGeometries[oNextSerial.lNr] == nullptr || poParent->papoBuiltGeometries[oNextSerial.lNr]->getGeometryType() != wkbPoint )

Perhaps you would als want to apply, as a precaution, a similar fix for the curve and multiPoint cases just above.

Reply:
Thanks for the suggestion. Will add that later today...